client/InvoiceTickets.js

```diff
@@ -7,11 +16,7 @@ Template.registerHelper('equals', function(a, b){
 
 Template.InvoiceTickets.helpers({
   tickets: function(){
-    return InvoiceTickets.byTimeRange(
-      FlowRouter.getParam("filter"),
-      FlowRouter.getQueryParam("sortBy"),
-      FlowRouter.getQueryParam("sortOrder")
-    );
+    return InvoiceTickets.find();
```

Comment:
You also have to use `InvoiceTickets.byTimeRange` on the client! Otherwise the filter views won't work correctly. This is because Meteor is smart and keeps a cache of the documents on the client after you subscribed to them once before. In your case if you click on "month" and the client receives all these docs, then you click on "today" there will still be all the documents of the "month" visible. Short: you need to use the same query on client *and*  server :wink: on the server, it's to publish only a subset of docs to the client. On the client it's to only show the correct subset of documents in the UI.